### ISSUE TITLE
When trying to use MBWaypoint or MBRoute the application crash unexpectedly.

### ISSUE THREAD

**Issue by BoogieMAN2K** [NAXAM/mapboxnavigation-ios-binding#1]
It doesn't throw any error message or alert, the app just exit.

The issue occurs when using those classes on any piece of code even if the are not been actively used anywhere. I'm attaching a piece of code that is having that behavior [HomeController.zip](https://github.com/NAXAM/mapboxnavigation-ios-binding/files/1995848/HomeController.zip), when uncommenting the code the app crashes without any warning.

**Comment by tuyen-vuduc** [comment]
@BoogieMAN2K Have you tried to turn off linking?

**Comment by tuyen-vuduc** [comment]
Have you check device/simulator logs?

**Comment by BoogieMAN2K** [comment]
Yes, that's the strangest thing, the app leaves while on the splash screen. It doesn't even reach the screen with the code. I clean / delete the obj/bin folders thinking it was a problem with old code being reuse but it didn't work either. 

Nor the simulator nor the device shows anything on the logs.

**Comment by BoogieMAN2K** [comment]
Hi, any update on this. Do you need any additional information about the issue?

**Comment by stevenkangas** [comment]
I am seeing the same issue.  Any status on this?

**Comment by osca** [comment]
I'm facing the same issue and cannot work around it, because I need to split two direction calculation requests into a single route (api limitation of 25 coordinates). The app crashes immediately after creating a MBRoute or trying to change/set its coordinates property. Does somebody have any idea?

**Comment by devcodet** [comment]
I am seeing the same issue right where as well. 
The iOS app just exit without crash in Visual Studio and like @osca say "The app crashes immediately after creating a MBRoute or trying to change/set its coordinates property"
Any updates?

I'am working with:
Naxam.MapboxNavigation.iOS v. 0.16.0
Naxam.Mapbox.iOS v. 4.3.0.3
